tweetpage

Barylo opened the pull request:
Hi guys,
I've added:
SingleTweetPage (the page where you get when click on a tweet). On it you will see the back arrow button, your single tweet (I've borrowed from ArtemLigerko work), comment input and button, list of comments (also from ArtemLigerko work).

Please review it and let it be merged with the project.
![image](https://user-images.githubusercontent.com/89450161/185114527-d5f24d7e-96e0-4c9f-8b31-d4826a0cb1ba.png)

Barylo committed:
tweetpage

alexandrtovmach commented on package.json:
```diff
@@ -8,12 +8,18 @@
+    "@emotion/react": "^11.10.0",
+    "@emotion/styled": "^11.10.0",
+    "@mui/icons-material": "^5.8.4",
+    "@mui/material": "^5.10.0",
```
we're using bootstrap, so installig material UI it's bad idea

alexandrtovmach commented on package.json:
```diff
@@ -8,12 +8,18 @@
   "dependencies": {
+    "@emotion/react": "^11.10.0",
+    "@emotion/styled": "^11.10.0",
+    "@mui/icons-material": "^5.8.4",
```
we agreed to use bootstrap icons, so it shouldn't be here

alexandrtovmach commented:
resolve merge conflicts with `develop` branch

VitaliiDenysiuk commented on src/containers/CommentsList.tsx:
```diff
@@ -0,0 +1,26 @@
+      <SingleTweetComment />
+      <SingleTweetComment />
+      <SingleTweetComment />
+      <SingleTweetComment />
```
```suggestion
      <SingleTweetComment />
```

use loop here

VitaliiDenysiuk commented on src/pages/SingleTweet.tsx:
```diff
@@ -0,0 +1,224 @@
+  }
+`;
+
+interface IIcon {
```
```suggestion
interface IIcon {
```

VitaliiDenysiuk replied:
don't use "I" for naming interface

ArtemLigerko replied:
Why?

Barylo replied:
This is not my area, it's Artem's

VitaliiDenysiuk commented on src/pages/SingleTweetComment.tsx:
```diff
@@ -0,0 +1,243 @@
+  }
+`;
+
+interface IIcon {
```
```suggestion
interface IIcon {
```
you declarate such interface in two different module. Create it in one place and import it in seconde

Barylo replied:
This is Artem's component. And it is already in develop branch

KirillSerg commented on src/containers/TweetPage.css:
```diff
@@ -0,0 +1,15 @@
+.tweetHeader {
```
it might be better to use bootstrap component or styled component

Barylo replied:
Done

ArtemLigerko commented:
Branch naming must corresponds to Conventional Commits (like 'feature/tweetpage')

ArtemLigerko commented:
move files with styles in 'styles' folder, and we use 'styled-components' instead css

ArtemLigerko commented on src/containers/TweetPage.tsx:
```diff
@@ -0,0 +1,37 @@
+import { Grid, Paper, IconButton } from "@mui/material";
+
+import SingleTweet from "../pages/SingleTweet";
+// import { useHistory } from "react-router-dom";
```
remove lines with comments

Barylo replied:
Done

Mykyta-Pysarev commented on src/index.tsx:
```diff
@@ -1,6 +1,7 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
 import { BrowserRouter } from "react-router-dom";
+import "bootstrap/dist/css/bootstrap.min.css";
```
Remove import of styles. We already has it on top level of the app.

Mykyta-Pysarev commented on src/pages/SingleTweet.tsx:
```diff
@@ -0,0 +1,224 @@
+`;
+const Author = styled.div`
+  height: 48px;
+  margin-top: 12px;
```
Use bootstrap for margins and paddings. [bootstrap-spacing](https://getbootstrap.com/docs/4.0/utilities/spacing/)

Barylo replied:
done

Mykyta-Pysarev commented on src/pages/SingleTweetComment.tsx:
```diff
@@ -0,0 +1,243 @@
+  fillColor: string;
+}
+const Icon = styled.svg<IIcon>`
+  width: 18.75px;
```
Why such odd width and height? How monitor can properly show 0.75 of pixel? Round up, better use 'rem' instead of pixels.

Barylo replied:
it is Artem's component. And it is already in develop branch

Barylo committed:
fix: during mergin with develop

alexandrtovmach commented on src/containers/Home.tsx:
```diff
@@ -1,5 +1,17 @@
+// import { InputLine } from "../pages/InputLine";
+// import { CommentsList } from "./CommentsList";
+// import { TweetPage } from "./TweetPage";
```
remove comments

Barylo replied:
Done. You'll see it when I push

alexandrtovmach commented on src/containers/Home.tsx:
```diff
@@ -1,5 +1,17 @@
+
+      {/* <TweetPage />
+      <InputLine />
+      <CommentsList /> */}
```
remove comments

Barylo replied:
Done

Barylo committed:
fixed bugs

Barylo committed:
deleted comments

Barylo commented:
Done. Please review.

alexandrtovmach commented on package.json:
```diff
@@ -23,6 +23,7 @@
     "@types/react": "^18.0.17",
     "@types/react-dom": "^18.0.6",
     "bootstrap": "^5.2.0",
+    "bootstrap-icons": "^1.9.1",
```
check if it's required or can be deleted

Barylo replied:
done

alexandrtovmach commented on src/containers/TweetPage.tsx:
```diff
@@ -0,0 +1,37 @@
+          viewBox="0 0 16 16"
+        >
+          <path d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z" />
+        </svg>
```
use bootstrap icons as a components:
https://github.com/attech-org/leleka/blob/develop/src/components/SingleTweet.tsx#L169

Barylo replied:
done

alexandrtovmach commented on src/pages/SingleTweet.tsx:
```diff
@@ -0,0 +1,224 @@
+import { Dropdown } from "react-bootstrap";
```
remove this file, and use SingleTweet that available from develop

Barylo replied:
done

alexandrtovmach commented on src/pages/SingleTweetComment.tsx:
```diff
@@ -0,0 +1,243 @@
+import { Dropdown } from "react-bootstrap";
```
same

Barylo replied:
done

Barylo committed:
fix: delete bootstrap-icons and two redundent components

Barylo committed:
fix: some errors

Barylo committed:
Merge branch 'develop' into tweetpage

Barylo committed:
fix: debuged

Barylo committed:
fix: added data to TweetPage form MOCKS, have a bug in CommentsList component

Barylo committed:
Merge branch 'develop' of github.com:attech-org/leleka into develop

Barylo committed:
Merge branch 'develop' into tweetpage

Barylo commented:
I've added data to "const singleTweetData" because there were empty strings and nothing to render.
Added mockdata of  const singleTweetData to TweetPage.
Added mockdata of  const tweetCommentsData to CommentsList container - and here is a bug. Would you please to make a review of the code, something is missing in const tweetCommentsData
![image](https://user-images.githubusercontent.com/89450161/185745616-9d109822-9bbd-4306-a380-05c126698f32.png)

alexandrtovmach commented on src/containers/CommentsList.tsx:
```diff
@@ -0,0 +1,12 @@
+export const CommentsList: React.FC = () => {
+  return (
+    <div>
+      <SingleTweetComment {...tweetCommentsData} />
```
tweetCimmentsData it's array of objects with comment data, so all what you need is to render each object by passing it to singletweetcomment in map cycle

DavidKorystov replied:
Example:

![image](https://user-images.githubusercontent.com/94183891/185785272-0d436f56-fff4-438c-a23a-d1e8190cc69f.png)

Barylo committed:
fix: all bugs

Barylo committed:
Merge branch 'develop' of github.com:attech-org/leleka into develop

Barylo committed:
Merge branch 'develop' into tweetpage

alexandrtovmach commented on tash:
```diff
@@ -0,0 +1,111 @@
+[33mcommit 981a565a23020346ba2f7dc5b78c720e4a5e4fb3[m
```
something unexpected

Barylo replied:
have no idea

alexandrtovmach commented on src/containers/CommentsList.tsx:
```diff
@@ -0,0 +1,25 @@
+export const CommentsList: React.FC = () => {
+  return (
+    <div>
+      {tweetCommentsData.map((obj) => (
```
better to use proper names for arguments and variables, for example "comment"

alexandrtovmach commented on src/containers/CommentsList.tsx:
```diff
@@ -0,0 +1,25 @@
+          messegeDate={obj.messegeDate}
+          answerCount={obj.answerCount}
+          retweetCount={obj.retweetCount}
+          likeCount={obj.likeCount}
```
if all prop names match with keys, you can simply use spread operator:

```
<SingleTweetComment key={obj.id} {...obj} />
```

Barylo committed:
fix: repaired the view of props
